src/main/java/hudson/plugins/analysis/core/AbstractResultAction.java

```diff
@@ -104,15 +111,30 @@ public ToolTipProvider getToolTipProvider() {
      *
      * @return the associated build of this action
      */
-    public final AbstractBuild<?, ?> getOwner() {
+    @WithBridgeMethods(value=AbstractBuild.class, adapterMethod="getAbstractBuild")
+    public final Run<?, ?> getOwner() {
         return owner;
     }
 
     @Override
-    public final AbstractBuild<?, ?> getBuild() {
+    @WithBridgeMethods(value=AbstractBuild.class, adapterMethod="getAbstractBuild")
+    @Deprecated
```

Comment:
I might be missing something, but why are you introducing `@Deprecated` on this method in this PR? Is it overriding/implementing a deprecated method? If so, this is gratuitous as the annotation is inherited anyway.